### PR TITLE
Fix some compiler warnings

### PR DESCRIPTION
**Pull request by AtariDreams**
Warnings about using the wrong type, wrong types being passed to printf-like functions, etc.

### CODE REVIEW

File: authfile.c [hunk 1]
```diff
@@ -442,9 +442,7 @@ sshkey_in_file(struct sshkey *key, const char *filename, int strict_type,
 int
 sshkey_check_revoked(struct sshkey *key, const char *revoked_keys_file)
 {
-	int r;
-
-	r = ssh_krl_file_contains_key(revoked_keys_file, key);
+	int r = ssh_krl_file_contains_key(revoked_keys_file, key);
```
Comment (daztucker) [inline authfile.c:445]: style guide (https://man.openbsd.org/style.9) says not to use function calls as variable initializers.

File: authfile.c [hunk 2]
```diff
@@ -474,10 +472,10 @@ sshkey_check_revoked(struct sshkey *key, const char *revoked_keys_file)
 int
 sshkey_advance_past_options(char **cpp)
 {
-	char *cp = *cpp;
+	char *cp;
 	int quoted = 0;
 
-	for (; *cp && (quoted || (*cp != ' ' && *cp != '\t')); cp++) {
+	for (cp = *cpp; *cp && (quoted || (*cp != ' ' && *cp != '\t')); cp++) {
```
Comment (daztucker) [inline authfile.c:478]: what warning does this fix?

File: openbsd-compat/base64.c [hunk 3]
```diff
@@ -144,8 +144,8 @@ b64_ntop(u_char const *src, size_t srclength, char *target, size_t targsize)
 		srclength -= 3;
 
 		output[0] = input[0] >> 2;
-		output[1] = ((input[0] & 0x03) << 4) + (input[1] >> 4);
```
Comment (daztucker) [inline openbsd-compat/base64.c:147]: what's warning is this supposed to fix?

**Review by daztucker — COMMENTED** [review]
There are unrelated changes in the PR, eg changes to the autoconf files.

**Comment by AtariDreams** [comment]
@daztucker Addressed!

**Comment by AtariDreams** [comment]
> There are unrelated changes in the PR, eg changes to the autoconf files.

The restrict addition is so that __restrict does nothing on compilers that do not support it, while having the restrict keyword on platforms that do.

**Comment by AtariDreams** [comment]
> There are unrelated changes in the PR, eg changes to the autoconf files.

I removed the bzero check since I replaced it with memset and then reran autoreconf

File: openbsd-compat/base64.c (hunk 3 above)
Comment (daztucker) [reply openbsd-compat/base64.c:147]: You didn't answer my question.

File: openbsd-compat/bsd-misc.c [hunk 4]
```diff
@@ -387,14 +387,6 @@ llabs(long long j)
...
 #endif
 
-#ifndef HAVE_BZERO
-void
-bzero(void *b, size_t n)
-{
-	(void)memset(b, 0, n);
-}
-#endif
-
```
Comment (daztucker) [inline openbsd-compat/bsd-misc.c:397]: bzero is still used in some places (eg the compat library, netcat in the tests) so this will break on platforms that don't have bzero.  It's also unrelated to the purported purpose of this commit so doesn't belong in this commit (in addition to being pointless churn).

File: regress/netcat.c [hunk 5]
```diff
@@ -495,7 +495,7 @@ main(int argc, char *argv[])
 					sv = NULL;
 				else {
 					sv = getservbyport(
-					    ntohs(atoi(portlist[i])),
+					    ntohs((int)strtol(portlist[i], NULL, 10)),
```
Comment (daztucker) [inline regress/netcat.c:498]: what warning is this fixing?  atoi already returns an int which is what's needed.

File: regress/unittests/test_helper/test_helper.c [hunk 6]
```diff
@@ -392,7 +392,7 @@ tohex(const void *_s, size_t l)
 
 	assert(r != NULL);
 	for (i = j = 0; i < l; i++) {
-		r[j++] = hex[(s[i] >> 4) & 0xf];
+		r[j++] = hex[s[i] >> 4];
```
Comment (daztucker) [inline regress/unittests/test_helper/test_helper.c:395]: what warning is this fixing?

it's also probably wrong on platforms that don't have an integral 8-bit type (Cray is the only example I know of and those are pretty rare these days, but it's allowed by the standards).

File: sc25519.c [hunk 7]
```diff
@@ -166,7 +166,7 @@ int sc25519_lt_vartime(const sc25519 *x, const sc25519 *y)
 
 void sc25519_add(sc25519 *r, const sc25519 *x, const sc25519 *y)
 {
-  int i, carry;
+  crypto_uint32 i, carry;
```
Comment (daztucker) [inline sc25519.c:169]: What warning is this fixing?

This code is taken entirely from its upstream (in supercop) and is only ever used for small integer loops so we'd need a really good reason to want to change it.

**Review by daztucker — CHANGES_REQUESTED** [review]
You've still got unrelated changes in this PR (all of the changes to config.guess, config.sub, install-sh).  You're also mixing different classes of changes in a single commit that are  (eg replacing bzero, which is completely unjustified by the stated goal of this PR).

File: openbsd-compat/base64.c (hunk 3 above)
Comment (AtariDreams) [reply openbsd-compat/base64.c:147]: > Will roll back. Meant for another PR

**Comment by AtariDreams** [comment]
Done @daztucker

**Comment by AtariDreams** [comment]
Ok, it is all ready now!

**Comment by AtariDreams** [comment]
@daztucker Thoughts on this?
